test/Integration.t.sol

```diff
@@ -0,0 +1,311 @@
+
+import { CapAutomator } from "../src/CapAutomator.sol";
+
+contract CapAutomatorIntegrationTestsBase is Test {
+
+    address public constant POOL_ADDRESSES_PROVIDER = 0x02C3eA4e34C0cBd694D2adFa2c690EECbC1793eE;
+    address public constant POOL                    = 0xC13e21B648A5Ee794902342038FF3aDAB66BE987;
+    address public constant POOL_CONFIG             = 0x542DBa469bdE58FAeE189ffB60C6b49CE60E0738;
+    address public constant DATA_PROVIDER           = 0xFc21d6d146E6086B8359705C8b28512a983db0cb;
+    address public constant ACL_MANAGER             = 0xdA135Cd78A086025BcdC87B038a1C462032b510C;
+    address public constant SPARK_PROXY             = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+    address public constant WETH                    = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant WBTC                    = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+
+    address public user;
+
+    address[] assets;
+
+    CapAutomator public capAutomator;
+
+    IACLManager aclManager = IACLManager(ACL_MANAGER);
+    IPool       pool       =       IPool(POOL);
+
+    function setUp() public {
+        vm.createSelectFork(getChain("mainnet").rpcUrl, 18_721_430);
+
+        capAutomator = new CapAutomator(POOL_ADDRESSES_PROVIDER);
+
+        capAutomator.transferOwnership(SPARK_PROXY);
+
+        vm.prank(SPARK_PROXY);
+        aclManager.addRiskAdmin(address(capAutomator));
+
+        assets = pool.getReservesList();
+
+        user = makeAddr("user");
+    }
+
+}
+
+contract GeneralisedTests is CapAutomatorIntegrationTestsBase {
+
+    using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+    using WadRayMath           for uint256;
+
+    function test_E2E_increaseBorrow() public {
+        for(uint256 i; i < assets.length; i++) {
+            DataTypes.ReserveData memory reserveData = pool.getReserveData(assets[i]);
+
+            uint256 preIncreaseBorrowCap = reserveData.configuration.getBorrowCap();
+            if (preIncreaseBorrowCap == 0) {
+                continue;
+            }
+
+            uint256 currentBorrow = ERC20(reserveData.variableDebtTokenAddress).totalSupply() / 10 ** ERC20(reserveData.variableDebtTokenAddress).decimals();
+
+            uint256 preIncreaseBorrowGap = preIncreaseBorrowCap - currentBorrow;
+
+            uint256 newMaxCap = preIncreaseBorrowCap * 2;
+            uint256 newGap    = preIncreaseBorrowGap * 2;
+
+            vm.prank(SPARK_PROXY);
+            capAutomator.setBorrowCapConfig({
+                asset:            assets[i],
+                max:              newMaxCap,
+                gap:              newGap,
+                increaseCooldown: 12 hours
+            });
+
+            capAutomator.exec(assets[i]);
+
+            uint256 postIncreaseBorrowCap = pool.getReserveData(assets[i]).configuration.getBorrowCap();
+
+            assertEq(postIncreaseBorrowCap, currentBorrow + newGap);
```

Comment:
I would include more asserts to all of these generalized tests checking that the cooldown is set correctly for increase and not changed for decrease.